ImapPromise.js

```diff
@@ -24,6 +24,20 @@ function connectAsync(){
 		imap.connect();
 	});
 }
+
+function getBoxesAsync() {
```

Comment:
The formatting in the original file is quite condensed. I didn't want to follow that and I'd rather format the rest of the code as well, but I can change the new code to the same format if desired.